spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -405,15 +411,23 @@ public boolean equals(Object other) {
       return false;
     }
 
-    // use only name in order to correctly invalidate Spark cache
+    // use name only unless branch/snapshotId is given in order to correctly invalidate Spark cache
```

Comment:
I think the new comment is a bit ambiguous. What about this?

```
// use table name, branch, and snapshot ID to correctly invalidate Spark cache
```

Or even drop the comment if the implementation is clear enough?